src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -98,6 +102,115 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _defaultAppearanceViewModel.IsDefault(true);
     }
 
+    void ProfileViewModel::LeftPadding(double value) noexcept
+    {
+        const hstring& padding = _GetNewPadding(PaddingDirection::Left, value);
+
+        Padding(padding);
+    }
+
+    double ProfileViewModel::LeftPadding() const noexcept
+    {
+        return _GetPaddingValue(PaddingDirection::Left);
+    }
+
+    void ProfileViewModel::TopPadding(double value) noexcept
+    {
+        const hstring& padding = _GetNewPadding(PaddingDirection::Top, value);
+
+        Padding(padding);
+    }
+
+    double ProfileViewModel::TopPadding() const noexcept
+    {
+        return _GetPaddingValue(PaddingDirection::Top);
+    }
+
+    void ProfileViewModel::RightPadding(double value) noexcept
+    {
+        const hstring& padding = _GetNewPadding(PaddingDirection::Right, value);
+
+        Padding(padding);
+    }
+
+    double ProfileViewModel::RightPadding() const noexcept
+    {
+        return _GetPaddingValue(PaddingDirection::Right);
+    }
+
+    void ProfileViewModel::BottomPadding(double value) noexcept
+    {
+        const hstring& padding = _GetNewPadding(PaddingDirection::Bottom, value);
+
+        Padding(padding);
+    }
+
+    double ProfileViewModel::BottomPadding() const noexcept
+    {
+        return _GetPaddingValue(PaddingDirection::Bottom);
+    }
+
+    winrt::hstring ProfileViewModel::_GetNewPadding(PaddingDirection paddingDirection, double newPaddingValue) const
+    {
+        std::array<double, 4> values{};
+        std::wstring_view remaining{ Padding() };
+        uint32_t paddingIndex = static_cast<uint32_t>(paddingDirection);
+
+        try
+        {
+            for (uint32_t index = 0; !remaining.empty() && index < values.size(); ++index)
+            {
+                const std::wstring token{ til::prefix_split(remaining, L',') };
+                auto curVal = std::stod(token);
+
+                if (paddingIndex == index)
+                {
+                    curVal = newPaddingValue;
+                }
+
+                values[index] = curVal;
+            }
+        }
+        catch (...)
+        {
+            values.fill(0);
+            LOG_CAUGHT_EXCEPTION();
+        }
+
+        const auto result = fmt::format(FMT_COMPILE(L"{:.6f}"), fmt::join(values, L","));
+
+        return winrt::hstring{ result };
+    }
+
+    double ProfileViewModel::_GetPaddingValue(PaddingDirection paddingDirection) const

```

Comment:
My only concern is that this pair of functions will mis-handle 1-part and 2-part paddings:

padding `8` should be parsed as `8,8,8,8`; padding `8,4` should be parsed as `8,4,8,4`

everything else, I'm totally ready to sign off!